Deprecate torch.matrix_power in favor of torch.linalg.matrix_power

heitorschueroff opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53538 Deprecate torch.matrix_power in favor of torch.linalg.matrix_power**
* #52608 Added torch.linalg.matrix_power

Differential Revision: [D27261531](https://our.internmc.facebook.com/intern/diff/D27261531)

heitorschueroff committed:
Deprecate torch.matrix_power in favor of torch.linalg.matrix_power

[ghstack-poisoned]

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"




[ghstack-poisoned]

mruberry reviewed (commented):
What should we do about
 https://github.com/pytorch/pytorch/blob/a76b4736dbf6c91f11e57c2141104452d0c2fc4c/test/test_linalg.py#L5396

?

Should it be removed? Or updated to call linalg.matrix_power()?

mruberry commented:
I think we can make torch.matrix_power() an alias of torch.linalg.matrix_power()? We have support for aliases in OpInfos, too. And then the docs can point torch.matrix_power() to torch.linalg.matrix_power().

heitorschueroff commented:
> What should we do about
> https://github.com/pytorch/pytorch/blob/a76b4736dbf6c91f11e57c2141104452d0c2fc4c/test/test_linalg.py#L5396
> 
> ?
> 
> Should it be removed? Or updated to call linalg.matrix_power()?

I removed it because the new functions cover the same tests and more.

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"




[ghstack-poisoned]

heitorschueroff commented:
@mruberry I added the out parameter to `torch.matrix_power` and made it into an alias. I fixed some bugs with the alias testing as well.

mruberry commented on torch/_tensor_docs.py:
```diff
@@ -2270,11 +2270,12 @@ def callable(a, b) -> number
 matrix_power(n) -> Tensor
 
-See :func:`torch.matrix_power`
+.. note:: :meth:`~Tensor.matrix_power` is deprecated, use :func:`torch.linalg.matrix_power` instead.
```
This can still just point to the torch.matrix_power docs like the other methods

mruberry commented on torch/_torch_docs.py:
```diff
@@ -5066,37 +5066,12 @@ def merge_dicts(*dicts):
 
-            [[  0.2468,  -6.7168],
-             [  2.0774,  -0.8187]]])
+Alias for :func:`torch.linalg.matrix_power`
```
Period at end of this sentence (yes it's super weird that the convention in _torch_docs is not to put a period at the end of its reference sentence; I don't know how that got started).

mruberry reviewed (approved):
Cool! Just two minor docs suggestions.

The merge conflicts are odd and unlikely to be related to your changes. I've hit similar issues in the past, too. Importing, rebasing in jellyfish, and exporting might get the PR CI going.

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Deprecate torch.matrix_power in favor of torch.linalg.matrix_power"


Differential Revision: [D27261072](https://our.internmc.facebook.com/intern/diff/D27261072)

[ghstack-poisoned]
